### PR TITLE
Route manager

### PR DESCRIPTION
**Pull request by Rygone**
Add a Activity to manage the route

### CODE REVIEW

File: app/src/main/java/com/github/sdpteam15/polyevents/model/entity/Zone.kt [hunk 1]
```diff
@@ -175,15 +176,23 @@ data class Zone(
...
+            list.remove(edges.start!!.toLatLng())
+        val mean = mean(list)
         for (l in getZoneCoordinates()) {
-            val AM = minus(mid, l[0])
+            val AM = minus(mean, l[0])
             val AB = minus(l[1], l[0])
-            val BM = minus(mid, l[1])
+            val BM = minus(mean, l[1])
             val BC = minus(l[2], l[1])
             //https://math.stackexchange.com/questions/190111/how-to-check-if-a-point-is-inside-a-rectangle
```
Comment (Cell00phane) [inline app/src/main/java/com/github/sdpteam15/polyevents/model/entity/Zone.kt:193]: Maybe put this the link into the method documentation Robin, and add a doc stating briefly what the method does.

File: app/src/main/java/com/github/sdpteam15/polyevents/model/map/LatLngOperator.kt [hunk 2]
```diff
@@ -51,6 +51,26 @@ object LatLngOperator {
...
+        var nbr = 0
+        for (point in points) {
+            latitude += point.latitude
+            longitude += point.longitude
+            nbr += 1
+        }
+        return if (nbr != 0)
+            LatLng(latitude / nbr, longitude / nbr)
+        else
+            LatLng(0.0, 0.0)
```
Comment (Cell00phane) [inline app/src/main/java/com/github/sdpteam15/polyevents/model/map/LatLngOperator.kt:71]: Nice, didn't know we could do that in Kotlin

File: app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/MapsFragment.kt [hunk 3]
```diff
@@ -128,7 +128,7 @@ class MapsFragment(private val mod: MapsFragmentMod) : Fragment(),
 
                 addNewRouteButton.visibility = View.VISIBLE
                 removeRouteButton.visibility = View.VISIBLE
-                saveNewRouteButton.visibility = View.VISIBLE
+                saveNewRouteButton.visibility = View.INVISIBLE
```
Comment (Cell00phane) [inline app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/MapsFragment.kt:131]: You can also do GONE if you don't want it take space

**Review by Cell00phane — APPROVED** [review]

File: app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/MapsFragment.kt (hunk 3 above)
Comment (Rygone) [reply app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/MapsFragment.kt:131]: Yes but it will be set to visible later in the code

**Review by Feez2403 — APPROVED** [review]
Nice implementation

**Pull request merged**